setup.py

```diff
@@ -6,7 +6,9 @@
 
 if platform.python_implementation() != "PyPy":
     c_extension = Extension(
-        "backports.zoneinfo._czoneinfo", sources=["lib/zoneinfo_module.c"],
+        "backports.zoneinfo._czoneinfo",
+        sources=["lib/zoneinfo_module.c"],
+        extra_compile_args=["-std=c99"],
```

Comment:
I'm not quite sure how to do this in a cross-platform way, because this works for `gcc` and `clang` but not for MSVC, which warns about an unknown argument (though the command still succeeds).

Reply:
Here's some prior art: https://github.com/python/cpython/commit/0b7829e089f9f93e1387f240ae03d42ac7cd77c7#diff-2e01a9f45ceaa02248341ccd350bc7c0

MSVC is primarily meant to be a C++ compiler so it only has c++ mode switches, the standard of C it implements is basically C99 I think. So just leaving it without a flag should be fine.

Reply:
Does that work if someone is using cygwin or mingw or something?

Reply:
Good question, apparently `mingw` overrides `sys.platform` to be `mingw` when using gcc so it should work there:
https://github.com/msys2/MINGW-packages/blob/abd06ca92d876b9db05dd65f27d71c4ebe2673a9/mingw-w64-python2/0410-MINGW-build-extensions-with-GCC.patch#L54

Not sure about cygwin though I'd imagine it also overrides `sys.platform` for this reason.

Reply:
Hm, looks like mingw and cygwin return `mingw` and `cygwin`, respectively. I'm not familiar enough with Windows to know if there is a `gcc` for that platform, but I guess it's pretty safe to do it this way now, I suppose.

I guess the real question is how important is this. It seems like the worst thing that happens in the common case is that it raises a warning on Windows, but if someone does have a Windows system out there running `gcc`, I think this turns it into a hard failure.

I guess I could do the `sys.platform` check and then as a fallback put the disabling pragma under a check against `__STDC_VERSION__ >= 199901L` (since that's the only part that *requires* C99.

Reply:
While there's no way to run `gcc` without mingw or cygwin, `clang` can actually be built natively and made available available on Windows. Though even there they provide a utility to make the toolchain compatible with MSVC: https://clang.llvm.org/docs/MSVCCompatibility.html#msvc-compatibility

I think this might not be worth handling with the preprocessor directive, may as well wait to see if anyone brings it up as a real problem.

Reply:
The `-std=c99` is really only needed on `gcc` because the only C99 feature we're using is `_Pragma`, which is only in the GCC branch of the conditional compile, so I suppose we're safe.

Reply:
`clang` sets `__GNUC__` since it implements `gcc` compiler extensions.